[WIP]チャット画面の実装

takuma10Mi opened the pull request:
#what
viw画面のマークアップを行いました。
htmlで全体を作りその中にheaderとcontentsのブロックを作りました。
文字やマークを全てブロックとして考えて配置しました。

#why
チャット機能は本アプリケーションの根幹機能なため。

Commit:
index scss

チャット画面の実装

Commit:
チャット画面の実装

チャット画面の実装

miyagit commented on README.md:
```diff
@@ -1,24 +1,48 @@
+### Association
+- belongs_to :group
+- belongs_to :user
+- belongs_to :messages
```
こちら必要ないですね。

miyagit commented on README.md:
```diff
@@ -1,24 +1,48 @@
+### Association
+- has_many :messages
+- has_many :groups
+
```
中間テーブルへのアソシエーションを記述しましょう。

takuma10Mi replied:
has_many : groups_userを追加しました。

miyagit commented on README.md:
```diff
@@ -1,24 +1,48 @@
+|user_id|integer|null: falce, foreign_key: true|
+|group_id|interger|null: falce, foreign_key: true|
+
+### Asociation
```
中間テーブルへのアソシエーションを記述しましょう。

takuma10Mi replied:
has_many : groups_userを追加しました。

ky3090 replied:
多対多のアソシエーションについて、もう少し調べてみましょう！

miyagit reviewed (commented):
belongs_toは基本的に単数形にして、has_manyは複数形にしましょう。

Commit:
Update README.md

修正しました。レビューお願いします。

ky3090 commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,131 @@
+  width: calc(100vw - 300px);
+  float:right;
+  background: #fafafa;
+  // padding: 0 40px;
```
不要なコメントアウトは削除しましょう！

takuma10Mi replied:
削除しました。

ky3090 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,4 @@
+@import "reset";
+@import "messages";
+// @import "variable.scss";
```
もし、こちらも使用しないのであれば、削除しましょう！

takuma10Mi replied:
削除しました。

ky3090 commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,10 @@
+!!!
+%html
+  %head
+    %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
=>(ハッシュロケット)は使わないようにしましょう！

takuma10Mi replied:
%meta{content : "text/html",charset:"UTF-8", http-equiv: "Content-Type"}/
=>をけして：に変えました。

ky3090 commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,131 @@
+  float:left;
+
+  &__edit{
+    // height: 70px;
```
ここのコメントアウトも不要であれば削除しましょう！

ky3090 commented on README.md:
```diff
@@ -1,24 +1,50 @@
+
+### Asociation
+- has_many :user
+- has_many :group
```
ここのアソシエーションの指定、間違えています。
messageとgroup、userの関係をもう一度再考してみましょう！

takuma10Mi replied:
- belongs_to :group
- belongs_to :user
を追加してidを取得しました。

i-date commented:
修正分がまだ反映（push）されていないようですので、pushお願いします^^

ky3090 commented:
まだ投稿が反映されていないですね...

Commit:
指摘していただいたところを

修正しました。

ky3090 commented on README.md:
```diff
@@ -1,24 +1,49 @@
+### Asociation
+- has_many :user
+- has_many :message
+- has_many :groups_users
```
多対多のアソシエーションについて、もう一度調べてみましょう！

ky3090 commented:
マークアップもお済みでしたら、viewをスクショ画像でいただけますと幸いです！

takuma10Mi commented:
https://gyazo.com/1a5f3f0b027851ef59fce99658a53b13
スクショしました。確認お願いします。

Commit:
スクショをしました。

確認お願いします。

miyagit commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,10 @@
+!!!
+%html
+  %head
+    %meta{content: "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
ハッシュロケットを直しましょう。
文字列のハッシュをやめてシンボルにした方が良い理由は下記の記事に書いてあるので、参考にしましょう。

https://www.pikawaka.com/ruby/hash

takuma10Mi replied:
昨日、テックコネクトでメンターさんにcontentの方はシンボルに変えられるけど"http-equiv" => "Content-Type"は変えられないと説明されたのですがシンボルに変えるとシンタックスエラーで；か（）を入れてくださいとエラー文が出ます。

https://qiita.com/nilinomiya1222/items/cac9beff46d6a94be06a　の
hamlでハッシュロケットを使わない記法について書きたい

で調べましたが解決がわからなかったです。すみません。

i-date replied:
`"http-equiv" => "Content-Type"`は以下の2点の記述方法を試してみてください。
特に前者の方がいいと思います。
- `http: { equiv: { "Content-Type" } }`
- `"http-equiv": "Content-Type"`

takuma10Mi replied:
修正しました。

Commit:
Update application.html.haml

昨日、テックコネクトでメンターさんにcontentの方はシンボルに変えられるけど"http-equiv" => "Content-Type"は変えられないと説明されたのですがシンボルに変えるとシンタックスエラーで；か（）を入れてくださいとエラー文が出ます。

Commit:
Update application.html.haml

修正しました。

i-date reviewed (approved):
修正ありがとうございます！LGTMとさせていただきます^^

i-date commented:
マージ後にすみません、以下ご参考までに。
`http-equiv: "Content-Type"`でエラーが生じるのは`-`が原因です。
- `"http-equiv": "Content-Type"`ではダブルクオートで囲むことで`http-equiv`を文字列化してあげてからシンボル化しています。
- `http: { equiv: { "Content-Type" } }`はエラーになったと思いますが、記述ミスでした。
`http: { equiv: "Content-Type" }`でエラーが起きないようになると思います。
